scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/S3ConnectorDownloader.java

```diff
@@ -0,0 +1,250 @@
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.AmazonS3URI;
+import com.amazonaws.services.s3.internal.ServiceUtils;
+import com.amazonaws.services.s3.transfer.Download;
+import com.amazonaws.services.s3.transfer.MultipleFileDownload;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+import com.amazonaws.util.AwsHostNameUtils;
+
+
+/**
+ * Import file(s) from an S3 server using the Import_from_S3 task.
+ * This task can be launched with parameters.
+ *
+ * @author The ProActive Team
+ */
+public class S3ConnectorDownloader extends JavaExecutable {
+
+    private String s3LocalRelativePath;
+
+    private String s3Url;
+
+    private String bucketName = null;
+
+    private String keyName = null;
+
+    private String host = null;
+
+    private String scheme;
+
+    private String accessKey;
+
+    private String secretKey;
+
+    private boolean pathStyle = true;
+
+    private static final String S3_LOCAL_RELATIVE_PATH = "s3LocalRelativePath";
+
+    private static final String S3_URI = "s3Url";
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+
+        if (args.containsKey(S3_URI) && !args.get(S3_URI).toString().isEmpty()) {
+            s3Url = args.get(S3_URI).toString();
+            parseAmazonS3URI(s3Url);
+        } else {
+            throw new IllegalArgumentException("You have to specify a valid s3URI. Empty value is not allowed.");
+        }
+
+        if (args.containsKey(S3_LOCAL_RELATIVE_PATH) && !args.get(S3_LOCAL_RELATIVE_PATH).toString().isEmpty()) {
+            s3LocalRelativePath = args.get(S3_LOCAL_RELATIVE_PATH).toString();
+        } else {
+            //Default value is getLocalSpace() because it will always be writable and moreover can be used to transfer files to another data space (global, user)
+            s3LocalRelativePath = getLocalSpace();
+        }
+
+        accessKey = getThirdPartyCredential("S3_ACCESS_KEY");
+        secretKey = getThirdPartyCredential("S3_SECRET_KEY");
+        if (accessKey == null || secretKey == null) {
+            throw new IllegalArgumentException("You first need to add your s3 username and password (S3_ACCESS_KEY, S3_SECRET_KEY) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException {
+        List<String> filesRelativePathName = new ArrayList<>();
+
+        boolean s3PathIsPrefix = (keyName.lastIndexOf('/') == keyName.length() - 1);
+
+        File f = new File(s3LocalRelativePath);
+
+        // If the path already exists, print a warning.
+        if (f.exists()) {
```

Comment:
can you wrap this block in a separate method called "createDirIfNotExists(s3LocalRelativePath)"